### ISSUE TITLE
Login issue: Nothing happens after selecting "Sign in with github"

### ISSUE THREAD

**Issue by TimeScience** [baolong281/jupyter-copilot#23]
When I select "Copilot: Sign in with github" nothing happens.
This is jupyter 7.2.2 running locally via pycharm. I'm logged into github on the browser and have enabled pop-ups and all other browser tab settings that I can think of. Any suggestions (might

[copilot-jupyter-issue.webm](https://github.com/user-attachments/assets/4e6d696c-e7fe-4c4f-bcf9-3ade4aaf4fc0)

Note after further exploration console, this seems to potentially be the same as [Issue #16](https://github.com/baolong281/jupyter-copilot/issues/16) (but I already half submitted this issue so it seems easier to leave it here... feel free to merge with #16)

seems like maybe a poorly formed URL is causing it?

![image](https://github.com/user-attachments/assets/0b036310-7124-4910-83c9-a6f6f707070a)

**Comment by TimeScience** [comment]
looks like the server is running?
![image](https://github.com/user-attachments/assets/e74fcfd0-96f9-4bb7-86a4-4d260a229d43)


Some more output from the error:

![image](https://github.com/user-attachments/assets/909ade56-1de8-494a-99d1-a7bc10dd3eb0)

**Comment by jordloresj** [comment]
I have the same problem, and it looks like the server is running too

**Comment by Polanict** [comment]
Same problem here as well.

```
PS C:\Users\polan\PycharmProjects> jupyter labextension list            
JupyterLab v4.3.2
C:\Users\xxx\miniconda3\envs\xxx\share\jupyter\labextensions
        jupyterlab_pygments v0.3.0 enabled ok (python, jupyterlab_pygments)
        jupyter_copilot v0.1.1 enabled ok (python, jupyter_copilot)
        @jupyter-notebook/lab-extension v7.0.6 enabled ok
        @jupyter-widgets/jupyterlab-manager v5.0.9 enabled ok (python, jupyterlab_widgets)
```

**Comment by pls331** [comment]
Got the same problem here. On Windows.

**Comment by kamiradi** [comment]
Got the same problem when running jupyterlab inside docker.

**Comment by baolong281** [comment]
hi, can you try if [this solution](https://github.com/baolong281/jupyter-copilot/issues/16#issuecomment-2525461147) possibly works?

**Comment by marcelpauly** [comment]
I got the same problem when running Jupyter Notebook on macOS in Chrome. Couldn't get the solution from issue #16  working.

**Comment by ck090** [comment]
Okay, so I was getting the same issue and here is how it worked for me. I opened `jupyter lab` instead of `jupyter notebook` on Safari instead of any Chrome based browser and then clicked on `Open command pallete -> Copilot: sign in to GitHub` and the token magically popped open. I later authenticated it with GitHub, and it now works on my all my browsers including Arc which is my primary.

TLDR: Use a different browser (non-Chrome based) and/or try using `jupyter lab`, the token should pop up.

**Comment by hellenlima** [comment]
None of the solutions have worked for me.
Running JupyterLab on MacOS, Safari or Google Chrome.

**Comment by ck090** [comment]
> None of the solutions have worked for me. Running JupyterLab on MacOS, Safari or Google Chrome.

Can you try running Jupyter through this code?
```
from jupyter_copilot import _load_jupyter_server_extension
from jupyter_server.serverapp import ServerApp

if __name__ == "__main__":
    app = ServerApp.instance()

    # Configure ServerApp to allow all IP addresses and listen on the specified port
    app.ip = "0.0.0.0"  # Allow access from all IP addresses
    app.port = 8000  # Use port 8000
    app.open_browser = False  # Prevent the browser from opening automatically
    app.allow_root = True
    
    # Initialize the ServerApp instance and load extensions
    app.initialize()
    _load_jupyter_server_extension(app)

    # Start the application
    app.start()
```
This will start a new Jupyter server and then try testing out the above thing.

**Comment by TM-Wang** [comment]
I had the same issue and confirmed that the extension was successfully **enabled**. It wasn't until I checked the **Jupyter log** that I realized `I had missed installing Node`. After installing Node, everything worked very well.

**Comment by bjnobre** [comment]
I had the same issue. I havent' install node.js (apt install nodejs) and, then, restart container. After reopening, sign page was there!

**Comment by yusaiwen** [comment]
install nodejs in this environment will help. In my case:

mamba install nodejs

After the installation, relaunch the jupyter. The signin page will pop out.
